### PR TITLE
converted class based dashboard component to functional component

### LINKED ISSUE #8613: [UI] Change the `Dashboard` form class component to a functional component.
### Current Behavior
Dashboard is a class-based component

### Desired Behavior
Make the dashboard a functional component

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by EraKin575**
**Notes for Reviewers**

This PR fixes #8613 

changed class  based component to functional

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by EraKin575** [comment]
@aabidsofi19 please have a look

**Comment by Ghat0tkach** [comment]
@EraKin575 
 Let's discuss this on Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Review by Rajdip019 — APPROVED** [review]
LGTM. Amazing work @EraKin575

**Comment by aabidsofi19** [comment]
Great Job .  @EraKin575  but this pr also some of the same issues as in https://github.com/meshery/meshery/pull/8701#pullrequestreview-1623796520  . take a look

**Comment by EraKin575** [comment]
@Rajdip019 thanks!

**Comment by EraKin575** [comment]
@aabidsofi19 a lot of functions are unused as well as some states. Is it okay to just comment them out?

**Comment by Rajdip019** [comment]
> @aabidsofi19 a lot of functions are unused as well as some states. Is it okay to just comment them out?

Yes, you can clean them up. That's the best way.

**Review by Rajdip019 — APPROVED** [review]

**Comment by sudhanshutech** [comment]
@Rajdip019 @EraKin575 are all changes incorporated here?

File: ui/components/DashboardComponent.js [hunk 1]
```diff
@@ -660,265 +646,294 @@ class DashboardComponent extends React.Component {
...
+  //     {
+  //       name : "name",
+  //       label : "Component",
+  //       options : {
+  //         filter : false,
+  //         sort : true,
+  //         searchable : true,
+  //         setCellProps : () => ({ style : { textAlign : "center" } }),
+  //         customHeadRender : ({ index, ...column }, sortColumn) => {
+  //           return (
```
Comment (sudhanshutech) [inline ui/components/DashboardComponent.js:721]: why so much commented out code, are these have no use?

**Comment by Rajdip019** [comment]
> @Rajdip019 @EraKin575 are all changes incorporated here?

Yes

File: ui/components/DashboardComponent.js (hunk 1 above)
Comment (sudhanshutech) [reply ui/components/DashboardComponent.js:721]: if not then give description why they are commented

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

**Comment by sudhanshutech** [comment]
@EraKin575 will give a recording after making a build that all works fine snice there are lot of changes

File: ui/components/DashboardComponent.js (hunk 1 above)
Comment (EraKin575) [reply ui/components/DashboardComponent.js:721]: these functions weren't used anywhere in the component. That's why I commented them

Comment (EraKin575) [reply ui/components/DashboardComponent.js:721]: @sudhanshutech  these were functions that were there before I changed to functional component. I thought these functions would be used later with a newer version. So, I just commented these out and removed the pieces of code that were converted

**Comment by EraKin575** [comment]
@sudhanshutech working as expected :)

https://github.com/meshery/meshery/assets/100225222/b8882242-4e5d-4912-918d-37edbc1c1897

**Review by Rajdip019 — APPROVED** [review]

**Comment by EraKin575** [comment]
@aabidsofi19  please merge this if this looks good to do so

**Review by Rajdip019 — APPROVED** [review]

**Pull request merged**